### DISCUSSION TITLE
DIY REM Chem Controller vs Atlas Scientific Wi-Fi Pool Kit

### DISCUSSION THREAD

**Discussion by nate-marshall** [tagyoureit/nodejs-poolController#414] [REM] [answered]
Now that I have control of my SWG and [Pump ](https://github.com/tagyoureit/nodejs-poolController/issues/393)(thanks @rstrouse!) with njsPC/dashPanel, I am looking to build out a chem controller. I read through the [DIY REM Chem Controller wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller#sensors) which suggests using a WhiteBox Labs T3 Hat (out of stock in most places) for Raspberry Pi with AS Industrial probes. I also found their Wi-Fi Pool kit that comes with everything built out (circuits, sensors, enclosure, and, what looks like, an esp8266). There are two versions where one uses consumer-grade sensors ($440) and the other has a 3-in-1 (pH, ORP, temp) industrial sensor ($780).

Does anyone have any suggestions on a chem controller if you could build out your config all over again? I am not sure if it matters, but my pool is smaller (11k gal). Also, I am open to buiding and soldering a controller myself, but always happy to spend a few extra dollars to grab a plug-n-play controller like the wifi kit noted above. 

Pool Config:
11k gal Latham Claremont Fiberglass 
Hayward Aquarite SWG
Hayward VS900 TriStar Pump

**Comment by rstrouse** [comment]
Honestly after having been involved in creating 3 of these I wouldn’t change a thing.  It has been rock solid for me.  You should email Effrem at Atlas and ask them if they have an ETA.

Personally I would rather hook up separate carrier boards than the Wi-Fi kit.  There is much more control using REM and EZO circuits.  As for multi probe @tagyoureit has one of them.  It is just two probes in one.

**Comment by tagyoureit** [comment, marked as answer]
I would not recommend the Pool Kits for integration with REM/Nixie.  The sensors are controlled in the Pool kit through Thingspeak.  We control the EZO chips directly through the lower level i2c protocol.  You can't have both of them running at once and both talking to the probes.  I'd stick with the individual components.

Interestingly, DigiKey seems to have them in stock.  https://www.digikey.com/en/products/detail/atlas-scientific/WB-T3/16003039

**Comment by nate-marshall** [comment]
I appreciate the input on this! I grabbed a T3 hat from DigiKey as well as the other circuits and probes direct from Atlas Scientific. I was thinking about the multi probe today and thought it would be better to have the flexibility to change out just the one probe if only a single probe fails vs having to replace the whole 3-in-1 probe if only a single probe fails. Lastly, I grabbed an IPS Complete Flow Cell which should keep me busy for awhile until I start looking into using an acid tank. 🕺

**Comment by rstrouse** [comment]
Take your time drilling out the flow cell for 3/4"  If you are using a drill press make sure it doesn't have a bunch of runout.  One of my neighbors goobered one up on his drill press.  You can drill it by hand but make sure you have it solidly screwed down.

**Comment by nate-marshall** [comment]
I am making great progress with my REM Chem setup. I have my pH and ORP probes mounted, calibrated, and reporting. I bought [this flow cell](https://sunplay.com/products/ips-controllers-flow-cell-complete-fc100g) that has a flow switch built into it and I [configured it according to the second option in the wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller#wire-the-flow-switch) using the GPIO ports, but it doesn't seem to be reporting any data back. Is there a way to troubleshoot the GPIO ports in the REM? I tried increasing the log verbosity, but I didn't get much from those (attached).

I tried grabbing the gpio info directly from the system, but didn't have much luck. I also tried killing the REM processing thinking it was grabbing onto it, but nope.

```
nate@njspc:~$ echo 29 > /sys/class/gpio/export
-bash: /sys/class/gpio/export: Permission denied
nate@njspc:~$ sudo echo 29 > /sys/class/gpio/export
-bash: /sys/class/gpio/export: Permission denied
nate@njspc:~$ ls -la /sys/class/gpio/export
--w------- 1 root root 4096 Dec 31  1969 /sys/class/gpio/export
nate@njspc:~$ sudo cat /sys/class/gpio/export
cat: /sys/class/gpio/export: Permission denied
nate@njspc:~$ echo "can you tell I have no idea what I am doing?"
can you tell I have no idea what I am doing?
``` 

I see something happening during boot on REM. However, I don't really see anything else happening besides an odd `error: undefined` error. 

```
[4/25/2022, 4:20:29 PM] info: Initializing GPIO Pins 1
[4/25/2022, 4:20:29 PM] info: Configuring Mock Pin #29 Gpio #5:in on Header 1 Edge: both. {"activeLow":false,"reconfigureDirection":false}
[4/25/2022, 4:20:29 PM] info: Watching GPIO #5
```
[consoleLog(2022-04-25_16-20-28).log](https://github.com/tagyoureit/nodejs-poolController/files/8558131/consoleLog.2022-04-25_16-20-28.log)

Side note: I have both njsPC and REM running on the same pi with a T3 whitebox hat.

**Reply by rstrouse** [reply]
Run the following command.  You are running as a user who does not have access to GPIO.
`sudo adduser nate gpio`

Also, I see it says Configuring Mock Pin.  REM is not configuring that pin because it doesn't think the GPIO subsystem exists.

**Reply by nate-marshall** [reply]
It looks like I was already in those groups. I am running Ubuntu Server vs Raspbian. I have a feeling that I may want to backup my configs and flash my RPi with Raspbian. Thoughts?

```
nate@njspc:~$ sudo adduser nate gpio
The user `nate' is already a member of `gpio'.
nate@njspc:~$ groups
nate adm dialout cdrom sudo audio video plugdev games users input render netdev gpio spi i2c
```

I was running into some modprobe issues a couple of days ago and hacked a startup script for njspc and rem to run at @reboot. 

```
nate@njspc:~$ cat startAll.sh
#!/bin/bash

echo "modprobes"
sudo modprobe wire &
sudo modprobe w1-gpio &
sudo modprobe w1-therm &

echo "start njsPC"
cd ~/nodejs-poolController
git pull
npm install
npm start &
sleep 60

echo "start REM"
cd ~/relayEquipmentManager
git pull
npm install
npm start &
```

I tried installing and running raspi-config to enable the gpio interface, but that didn't seem to work. 

Off to make some backups...

**Reply by rstrouse** [reply]
I am not a Linux head so it is raspian for me.

**Reply by rstrouse** [reply]
Just run the server processes under the pi user and the permissions become less daunting.  Also resist the urge to use sudo for installations.  In marks the owner as root.

**Reply by nate-marshall** [reply]
I reimaged my pi with raspbian and reconfigured all of the things. I feel like it was best to use the raspbian os assuming that it is more widely used/tested. 

I tried a backup/restore, but ran into issues restoring REM so I went ahead with reconfiguring njspc and rem. Happy to share that the flow sensor is recognized now and added to my rem chem setup!

Super excited to start integrating mqtt w/ home assistant along with influxdb/grafana. I am following [this discussion](https://github.com/tagyoureit/nodejs-poolController/discussions/413) as it might be useful for some of the issues we have been working through with my hayward tristar pump. Where it's not responding back to the njspc. I am wondering if I would be able use my mqtt broker as a state machine of sorts. Using the flow sensor to validate whether the pump is on/off. Then, if in the on state, send intermittent speed updates to the pump and the mqtt broker.

All of this is a bit outside of this discussion, but I will fire over any updates that I might have in the other mqtt discussion.

**Reply by rstrouse** [reply]
njsPC will continue to send the appropriate speed updates but it is so odd that everything I read regarding the response from the HW pump is the same.  Yet yours is not responding.  Did you remove the pump control panel from the mix?  I read that if you hook it up to an omni the pump control panel has to be disconnected.

**Reply by nate-marshall** [reply]
I did disconnect the control panel from the pump and also tried swapping my rs485 usb adaptors. Still nothing back from the pump.

**Reply by tagyoureit** [reply]
When all else fails, try a new rs485 adapter and check the continuity of your wire to make sure there are no breaks.
